storm-core/src/clj/backtype/storm/command/blobstore.clj

```diff
@@ -0,0 +1,166 @@
+;; Licensed to the Apache Software Foundation (ASF) under one
+;; or more contributor license agreements.  See the NOTICE file
+;; distributed with this work for additional information
+;; regarding copyright ownership.  The ASF licenses this file
+;; to you under the Apache License, Version 2.0 (the
+;; "License"); you may not use this file except in compliance
+;; with the License.  You may obtain a copy of the License at
+;;
+;; http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+(ns backtype.storm.command.blobstore
+  (:import [java.io InputStream OutputStream])
+  (:use [backtype.storm config])
+  (:import [backtype.storm.generated SettableBlobMeta AccessControl AuthorizationException
+            KeyNotFoundException])
+  (:import [backtype.storm.blobstore BlobStoreAclHandler])
+  (:use [clojure.string :only [split]])
+  (:use [clojure.tools.cli :only [cli]])
+  (:use [clojure.java.io :only [copy input-stream output-stream]])
+  (:use [backtype.storm blobstore log util])
+  (:gen-class))
+
+(defn update-blob-from-stream
+  "Update a blob in the blob store from an InputStream"
+  [key ^InputStream in]
+  (with-configured-blob-client blobstore
+    (let [out (.updateBlob blobstore key)]
+      (try 
+        (copy in out)
+        (.close out)
+        (.createStateInZookeeper blobstore key)
```

Comment:
Shouldn't the blobstore handle this itself when the output is committed/closed?  It seems bad that the client has to ask for the ZK state to be updated separate from interacting with the output stream.